src/lib.rs

```diff
@@ -985,7 +983,7 @@ impl EGraph {
                         }
                         NormAction::LetLit(var, lit) => {
                             let value = self.eval_lit(lit);
-                            let etype = self.proof_state.type_info.infer_literal(lit);
+                            let etype = self.desugar.type_info.infer_literal(lit);
```

Comment:
I thought about making a `type_info` method on the `EGraph`, to not have to go through `desugar` every time, but opted to leave it as is for a minimal change.

Happy to refactor to that though, if that is preferred.

Reply:
I like the type_info method idea!

Reply:
I gave that a go, refactoring all non-mutable usages of `egraph.desugar.type_info`.